lecture_1/homework/app/models/clan.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class Clan < ApplicationRecord
+  has_many :warriors, dependent: :destroy
+
+  validates :name, presence: true, length: { maximum: 256 }
```

Comment:
Dodałaś unikalny indeks w bazie danych, ale zabrakło walidacji na modelu. To spowoduje, że w przypadku duplikatu nasza aplikacja się wywróci zamiast zwrócić błąd walidacji. Standardowo wszelkie unikalne indeksy czy constrainty typu `null: false` traktujemy jako ostatnią deskę ratunku - wolimy, żeby tego typu rzeczy zostały obsłużone przez railsy

Reply:
Hmm, trochę nie mogę tego znaleźć. Czy chodzi o 
`t.string :name, null: false, limit: 256`
przy tworzeniu tabelki w bazie danych?

Reply:
Zajrzyj do migracji `20190326_index_name_on_clans.rb`. W schemie objawia się to jako `t.index ["name"], name: "index_clans_on_name", unique: true`. Unikalne indeksy powinny mieć walidację `uniqueness` na modelu (tego brakuje), a wszystkie atrybuty, które mają `null: false` powinny mieć walidację `presence` (to akurat masz ok)

Reply:
Ach! Dzięki :grin: